forte/proc/aset2.py

```diff
@@ -0,0 +1,324 @@
+        psi4.core.print_out("\n  Warning: reference has no information on active orbital selections.")
+        psi4.core.print_out("\n  Please ensure that they are set manually in the inputs.")
+    psi4.core.print_out("\n  =====================================================================")
+    psi4.core.print_out("\n  ")
+    psi4.core.print_out("\n  ")
+    psi4.core.print_out("\n  ")
+
+    # In ASET(2) procedure, We will keep 2 different mo_space_info:
+    # mo_space_info: Active = AC + AA + AV, restricted = BO + BV, frozen = F
+    # mo_space_info_active: Active = AA, restricted = AC + AV, frozen = BO + BV + F
+    mo_space_info_active = forte.build_aset2_spaceinfo(ref_wfn, mo_space_info, options)
+
+    # Build total (A+B) integrals first
+    update_environment_options(options, B_list, ref_wfn)
+    ints_e = forte.make_ints_from_psi4(ref_wfn, options, mo_space_info)
+
+    # Compute ASET(mf) higher-level with mo_space_info_active(inner) and methods in options, -> E_high(origin)
+    energy_high = 0.0
+    if(do_aset_mf):
+        update_fragment_options(options, A_list, ref_wfn)
+        ints_f = forte.make_ints_from_psi4(ref_wfn, options, mo_space_info_active)
+        energy_high = forte_driver_fragment(state_weights_map, scf_info, forte.forte_options, ints_f, mo_space_info_active)
+
+    # Compute envrionment (B) amplitudes T1, T2 and energy corrections E_c^B, dress fragment ints
+    update_environment_options(options, B_list, ref_wfn)
+    energy_env, ints_dressed = forte_driver_environment(state_weights_map, scf_info, ref_wfn,  mo_space_info, mo_space_info_active, ints_e, options)
+
+    # Build dressed fragment integrals, starting from an ints object.
+    update_fragment_options(options, A_list, ref_wfn)
+    ints_f = None
+    options.set_bool('TRUNCATE_MO_SPACE', True)
+    mo_space_info_truncated = forte.build_aset2_spaceinfo(ref_wfn, mo_space_info, options)
+    # Build custom integrals for dressed H
+    #ints_f = forte.make_ints_from_psi4(ref_wfn, options, mo_space_info_active)
+    nmo = mo_space_info_truncated.size("ALL")
+    scalar = 0.0 
+    hcore = np.zeros((nmo, nmo))
+    eri_aa = np.zeros((nmo, nmo, nmo, nmo))
+    eri_ab = np.zeros((nmo, nmo, nmo, nmo))
+    eri_bb = np.zeros((nmo, nmo, nmo, nmo))
+    ints_f = forte.make_custom_ints(options, mo_space_info_truncated, scalar,
+                              hcore.flatten(), hcore.flatten(), eri_aa.flatten(),
+                              eri_ab.flatten(), eri_bb.flatten())
+    options.set_str('INT_TYPE', "FCIDUMP")
+
+    # Set the scalar for the newly-made integral
+    frz1 = ints_f.frozen_core_energy()
+    dressed_scalar = ints_dressed.scalar_energy()
+    scalar = dressed_scalar - frz1
+    #if(int_type_frag != "CONVENTIONAL") or (int_type_frag == "CONVENTIONAL" and frag_do_fci):
+    scalar += ints_e.nuclear_repulsion_energy()
+    ints_f.set_scalar(scalar)
+
+    # Build new ints for dressed computation
+    ints_f.set_ints_from_asints(ints_dressed)
```

Comment:
Why can't DSRG directly give `ints_f`?

Reply:
I think another reason is that by doing this we can keep the environment protocol on the python side and open to future changes. 
I will keep it here to see how everyone feels.

Reply:
OK, let's see how others feel about this. I think this just introduces unnecessary complications to both C and Python side. What I am asking for is just a simple function overloading on the C side.